### ISSUE TITLE
RandomRGBShift doesn't used correct values over batch

### ISSUE THREAD

**Issue by nmichlo** [kornia/kornia#1929]
### Describe the bug

`RandomRGBShift` does not work over a batch of images, this is actually caused by the `shift_rgb` function.

Specifically the old line in the current version `kornia==0.6.7`:
```python3
shifted = (image + torch.Tensor(shifts).view(1, 3, 1, 1).to(image)).clamp_(min=0, max=1)
```

I have noticed that this change has partially been implemented on main:
https://github.com/kornia/kornia/blob/349c6389f902f2c6f5c76f3dd0e9e1bf246a1fa1/kornia/enhance/shift_rgb.py#L17

However, the dim for stack has not been specified and thus it is equivalent to `dim=0`, if I am not mistaken this should instead be explicitly set to `dim=1`. Currently the resulting stacked shape would be `(3, -1)` instead of `(-1, 3)` which probably means if differing r,g,b values are specified, then the wrong values might be used. This is a silent error, because the view call relies on the number of elements which is correct in both cases, just the order is possibly wrong in the current version.
```python3
shifted = (image + torch.stack(shifts, dim=1).view(-1, 3, 1, 1).to(image)).clamp_(min=0, max=1)
```

### Reproduction steps

```bash
import kornia
import torch

aug = kornia.augmentation.RandomRGBShift(p=1.0)

image = torch.rand(3, 480, 640)
batch_1 = torch.rand(1, 3, 480, 640)
batch_8 = torch.rand(8, 3, 480, 640)

aug(image)
aug(batch_1)
aug(batch_8) # fails in 0.6.7, but probably wrong on main too
```

### Expected behavior

Should work over the batch?

### Environment

```shell
kornia==0.6.7
```

**Comment by nmichlo** [comment]
PR: https://github.com/kornia/kornia/pull/1930
